addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1339,18 +1340,22 @@ const Wysiwyg = Widget.extend({
         const mediaDialog = new weWidgets.MediaDialog(this, mediaParams, $node);
         mediaDialog.open();
 
-        mediaDialog.on('save', this, function (element) {
+        mediaDialog.on('save', this, async function (element) {
             if (!element) {
                 return;
             }
             // restore saved html classes
             if (params.htmlClass) {
-                element.className += " " + params.htmlClass;
+                const allClasses = element.className + " " + params.htmlClass;
+                element.className = Array.from(new Set(allClasses.split(" "))).join(" ");
```

Comment:
Why not `element.classList.add(params.htmlClass)` ? AFAIK it won't duplicate the added class.

Reply:
You are right , I didn't think of it. Thanks for the suggestion
Note : I added the split becase `htmlClass` can have more than one class and it breaks If I add it directly.

Reply:
Oh ok I see, then you might need to spread it like `element.classList.add(...params.htmlClass.split(" "));`.